docs/website/contents/for-developers/QueryVersioning.md

```diff
@@ -0,0 +1,77 @@
+
+In general, we can't assume that every node operator is running on the same/latest version, so a query might be available in one node, but not in another. At the beginning of every connection from a client to a node, the Network layer will negotiate a [`NodeToClientVersion`][n2c]. Before the client sends a query, it can now check that the negotiated version is not older than the associated version number of the query, and act accordingly (i.e. display an indicative error message, or use a different/fallback mechanism to do its job).
+
+## Implementation
+
+### Version types
+
+Our code does not use the negotiated [`NodeToClientVersion`][n2c] directly, but translates them first to a [`CardanoNodeToClientVersion`][cardano-n2c] and then to [`ShelleyNodeToClientVersion`][shelley-n2c].
+
+ - A [`ShelleyNodeToClientVersion`][shelley-n2c] is what is associated with every query in [`querySupportedVersion`][query-supported-version].
+ - A [`CardanoNodeToClientVersion`][cardano-n2c] specifies the [`ShelleyNodeToClientVersion`][shelley-n2c] for each era, or indicates that a specific [era][feature-table] is not supported. As an example, consider
+   ```haskell
+   pattern CardanoNodeToClientVersion10 :: BlockNodeToClientVersion (CardanoBlock c)
+   pattern CardanoNodeToClientVersion10 =
+       HardForkNodeToClientEnabled
+         HardForkSpecificNodeToClientVersion2
+         (  EraNodeToClientEnabled ByronNodeToClientVersion1
+         :* EraNodeToClientEnabled ShelleyNodeToClientVersion6
+         :* EraNodeToClientEnabled ShelleyNodeToClientVersion6
+         :* EraNodeToClientEnabled ShelleyNodeToClientVersion6
+         :* EraNodeToClientEnabled ShelleyNodeToClientVersion6
+         :* EraNodeToClientEnabled ShelleyNodeToClientVersion6
+         :* EraNodeToClientDisabled
+         :* Nil
+         )
+   ```
+   This tells us that in Shelley, Allegra, Mary, Alonzo and Babbage, we use `ShelleyNodeToClientVersion6`, and Conway is disabled. This means that no queries that were introduced in `ShelleyNodeToClientVersion7` can be used, and no queries in the Conway era are possible at all.
+
+   The mapping from `NodeToClientVersion`s to `CardanoNodeToClientVersion`s is [`supportedNodeToClientVersions`][supportedNodeToClientVersions]
+
+Additionally, all versions larger than a certain `NodeToClientVersion` (see [`latestReleasedNodeVersion`][latestReleasedNodeVersion]) are considered experimental, and are only offered in the initial version negotiation when a flag (currently, `ExperimentalProtocolsEnabled`) is set. Also see [`limitToLatestReleasedVersion`][limitToLatestReleasedVersion] and its call/usage sites.
+
+### Checks
+
+The supported query versions are only enforced in the [Shelley query encoder][shelley-encodeNodeToClient], ie in code run by clients. The server will answer queries even if the negotiated version is older than expected (which might be the case with a custom client implementation), following the robustness principle.
```

Comment:
Right now, it will always be "newer" as we never specifiy upper bounds for queries, but I will clarify this and add an example, cf https://github.com/input-output-hk/ouroboros-consensus/pull/273/files#r1283050163